crates/rspack_core/src/compiler/mod.rs

```diff
@@ -141,7 +144,9 @@ where
     // Fake this compilation as *currently* rebuilding does not create a new compilation
```

Comment:
This comment is stale?

Reply:
Not sure, not the author, I can't understand what `Fake this compilation as *currently* rebuilding` means, but `does not create a new compilation` is not stale 😂

Reply:
I dug into our codebase and found: https://github.com/branchseer/rspack/blob/bf1ac34f02b2676ac8806fcda449aa1543b45b04/crates/rspack_core/src/compiler/hmr.rs#L71. Just want to make sure the new compilation get updated to the `JsCompilation`(this would be better to add a comment somewhere, maybe in other PRs), otherwise it's a UB.